### ISSUE TITLE
Single-threaded mode to avoid lock issue

### ISSUE THREAD

**Issue by LaRiffle** [OpenMined/GridNode#18]
## Feature Description
I observe lock issues when running multiprocessing in my grid_node.py app, which then hangs. I think this is because the app itself is multi threaded. Is there a way of preventing this?

My command:
```
python grid_node.py --id charlie --host localhost --port 7602
```
App looks multi threaded:
<img width="847" alt="Capture d’écran 2020-06-01 à 08 50 00" src="https://user-images.githubusercontent.com/12446521/83383319-2746a180-a3e5-11ea-819f-e48518e932ec.png">

#### Running with
- Linux 20.04
- Python 3.8

**Comment by jmaunon** [comment]
First time I see this behavior. I am not capable to reproduce due to #11

**Comment by LaRiffle** [comment]
I guess pytorch might be responsible for this, Linux PyTorch and multiprocessing have a long story together.
I found a fix for my usecase, but usually 90% of this pbs can be solved with:
```
import torch
torch.set_num_threads(1)
```

**Comment by IonesioJunior** [comment]
@Benardi  Perhaps it can solve our [issue](https://github.com/OpenMined/PyGrid/issues/390).

**Comment by IonesioJunior** [comment]
Closed by #21
